tools/richlist.js

```diff
@@ -477,4 +543,17 @@ async function startSync() {
     makeRichList(latestBlock, 500, updateAccounts);
   }
 }
-startSync();
+
+web3.eth.getNodeInfo((err, nodeInfo) => {
+  let isParity = false;
+
+  console.log(`Node version = ${nodeInfo}`);
+  if (nodeInfo.split('/')[0].toLowerCase().includes('parity')) {
```

Comment:
```TypeError: Cannot read property 'split' of null
    at GetNodeInfoMethod.web3.eth.getNodeInfo (/home/ubuntu/explorer/routes/web3relay.js:54:16)
    at GetNodeInfoMethod._callee$ (/home/ubuntu/explorer/node_modules/web3-core-method/dist/web3-core-method.cjs.js:126:22)
    at tryCatch (/home/ubuntu/explorer/node_modules/regenerator-runtime/runtime.js:45:40)
    at Generator.invoke [as _invoke] (/home/ubuntu/explorer/node_modules/regenerator-runtime/runtime.js:271:22)
    at Generator.prototype.(anonymous function) [as next] (/home/ubuntu/explorer/node_modules/regenerator-runtime/runtime.js:97:21)
    at asyncGeneratorStep (/home/ubuntu/explorer/node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
    at _next (/home/ubuntu/explorer/node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
    at process._tickCallback (internal/process/next_tick.js:68:7)
TypeError: Cannot read property 'split' of null
    at GetNodeInfoMethod.web3.eth.getNodeInfo (/home/ubuntu/explorer/routes/web3relay.js:54:16)
    at GetNodeInfoMethod._callee$ (/home/ubuntu/explorer/node_modules/web3-core-method/dist/web3-core-method.cjs.js:126:22)
    at tryCatch (/home/ubuntu/explorer/node_modules/regenerator-runtime/runtime.js:45:40)
    at Generator.invoke [as _invoke] (/home/ubuntu/explorer/node_modules/regenerator-runtime/runtime.js:271:22)
    at Generator.prototype.(anonymous function) [as next] (/home/ubuntu/explorer/node_modules/regenerator-runtime/runtime.js:97:21)
    at asyncGeneratorStep (/home/ubuntu/explorer/node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
    at _next (/home/ubuntu/explorer/node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

Could you see the documentation please? https://web3js.readthedocs.io/en/v2.0.0-alpha.1/web3-eth.html

Reply:
More logs

```
Web3 has detected parity node configuration
Web3 has detected parity node configuration
{ Error: Cannot find module '../../methods/TraceFilterMethod'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/ubuntu/explorer/lib/trace/factories/MethodFactory.js:10:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Module._compile (/home/ubuntu/explorer/node_modules/pirates/lib/index.js:99:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Object.newLoader [as .js] (/home/ubuntu/explorer/node_modules/pirates/lib/index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/ubuntu/explorer/lib/trace/index.js:13:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30) code: 'MODULE_NOT_FOUND' }
{ Error: Cannot find module '../../methods/TraceFilterMethod'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/ubuntu/explorer/lib/trace/factories/MethodFactory.js:10:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Module._compile (/home/ubuntu/explorer/node_modules/pirates/lib/index.js:99:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Object.newLoader [as .js] (/home/ubuntu/explorer/node_modules/pirates/lib/index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/ubuntu/explorer/lib/trace/index.js:13:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30) code: 'MODULE_NOT_FOUND' }

/home/ubuntu/explorer/node_modules/solc/soljson.js:33
(Module.asmGlobalArg,Module.asmLibraryArg,buffer);var __GLOBAL__I_000101=Module["__GLOBAL__I_000101"]=asm["__GLOBAL__I_000101"];var __GLOBAL__sub_I_ABIFunctions_cpp=Module["__GLOBAL__sub_I_ABIFunctions_cpp"]=asm["__GLOBAL__sub_I_ABIFunctions_cpp"];var __GLOBAL__sub_I_ABI_cpp=Module["__GLOBAL__sub_I_ABI_cpp"]=asm["__GLOBAL__sub_I_ABI_cpp"];var __GLOBAL__sub_I_ASTJsonConverter_cpp=Module["__GLOBAL__sub_I_ASTJsonConverter_cpp"]=asm["__GLOBAL__sub_I_ASTJsonConverter_cpp"];var __GLOBAL__sub_I_AST_cpp=Module["__GLOBAL__sub_I_AST_cpp"]=asm["__GLOBAL__sub_I_AST_cpp"];var __GLOBAL__sub_I_ArrayUtils_cpp=Module["__GLOBAL__sub_I_ArrayUtils_cpp"]=asm["__GLOBAL__sub_I_ArrayUtils_cpp"];var __GLOBAL__sub_I_AsmAnalysis_cpp=Module["__GLOBAL__sub_I_AsmAnalysis_cpp"]=asm["__GLOBAL__sub_I_AsmAnalysis_cpp"];var __GLOBAL__sub_I_BMC_cpp=Module["__GLOBAL__sub_I_BMC_cpp"]=asm["__GLOBAL__sub_I_BMC_cpp"];var __GLOBAL__sub_I_CHC_cpp=Module["__GLOBAL__sub_I_CHC_cpp"]=asm["__GLOBAL__s
abort({"code":"MODULE_NOT_FOUND"}). Build with -s ASSERTIONS=1 for more info.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! etc-explorer@0.2.0 app: `node app.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the etc-explorer@0.2.0 app script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ubuntu/.npm/_logs/2019-10-20T13_38_55_565Z-debug.log
```

Reply:
Parity lib doesn't work either

```
(node:2209) UnhandledPromiseRejectionWarning: Error: Cannot find module '../../methods/ParityListAccountsMethod'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/ubuntu/explorer/lib/parity/factories/MethodFactory.js:10:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Module._compile (/home/ubuntu/explorer/node_modules/pirates/lib/index.js:99:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Object.newLoader [as .js] (/home/ubuntu/explorer/node_modules/pirates/lib/index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/ubuntu/explorer/lib/parity/index.js:13:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
(node:2209) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:2209) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Reply:
~~~
import {Network} from 'web3-net';
       ^

SyntaxError: Unexpected token {
~~~
the latest web3 extension uses the babel-* packages.
I guess you've missed `npm i`

Reply:
@hackmod I did `npm install` and I am using the latest LTS version of nodejs which is node 10

Reply:
thanks for your patience.
I've missed some `lib/methods` files that caused an error and now fixed.
thanks!